### PR TITLE
feat: ghcup

### PR DESCRIPTION
**Pull request by PhotonQuantum**

### CODE REVIEW

File: src/ghcup.rs [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+        let base_url = self.ghcup_base.trim_end_matches('/');
+
+        info!(logger, "fetching ghcup config...");
+        progress.set_message("downloading version file");
+        let yaml_url = get_yaml_url(base_url, &client).await?;
+        let last_modified = get_last_modified(&client, &yaml_url).await?;
+
+        let yaml_url = yaml_url.trim_start_matches("https://www.haskell.org/");
+        progress.finish_with_message("done");
+        Ok(vec![SnapshotMeta {
```
Comment (skyzh) [inline src/ghcup.rs:228]: ditto, use `force`.

File: src/main.rs [hunk 2]
```diff
@@ -149,6 +175,44 @@ fn main() {
                     index_bytes_pipe!(buffer_path, prefix)
                 );
             }
+            Source::Ghcup(source) => {
```
Comment (skyzh) [inline src/main.rs:178]: Would you please give an example of how data will be stored in filesystem, and bash commands to sync?

**Comment by skyzh** [comment]
Personally, I would prefer merging these three snapshots into one. We could figure out the source site by adding prefix to these configurations, and dispatch them to corresponding site in `get_object`.

**Comment by PhotonQuantum** [comment]
Pending for MergePipe

**Comment by PhotonQuantum** [comment]
Closed in favor of #78.

**Pull request closed without merging**